### ISSUE TITLE
Resize/redraw broken on macOS 12

### ISSUE THREAD

**Issue by geuzaine** [fltk/fltk#508]
@ManoloFLTK: I have just tested the latest FLTK master (4196daa77) with Gmsh on macOS 12 (Monterey), and it breaks window resizing/redrawing. For example, when clicking on some widgets nothing happens until I move the mouse and wait for about 1 second.

Going back to 59fc60ea4 ("Simpler code to support FLTK widgets in macOS OpenGL 3 windows") restores correct behavior. I haven't had time to bisect more.

**Comment by ManoloFLTK** [comment]
I'm afraid a test app showing the problem would be needed.

**Comment by geuzaine** [comment]
Ok I'll post something here if I succeed to reproduce the issue with a simple test app.

Meanwhile if you want to see the issue in the real app it's quite easy: on your mac run
```
git clone https://gitlab.onelab.info/gmsh/gmsh.git
cd gmsh && mkdir build && cd build && cmake -DENABLE_OS_SPECIFIC_INSTALL=1 .. && make -j 8 package
```
You can then open `gmsh-git-MacOSX.dmg` and run the app: resize the window and/or open Tools->Options and click on random widgets to see the issue.

**Comment by ManoloFLTK** [comment]
I tried to build gmsh following your instructions (adding "-DCMAKE_BUILD_TYPE=Debug -DCMAKE_OSX_ARCHITECTURES=arm64" to the cmake command) but obtain an executable called gmsh that runs no GUI, a nearly empty tar.gz file, and no .dmg file :
<img width="436" alt="image" src="https://user-images.githubusercontent.com/41016272/193425613-e960952b-b43d-4662-b91e-5b6c3b2459a6.png">

Where does the gmsh build procedure get the FLTK library ?

**Comment by ManoloFLTK** [comment]
I got gmsh built here now after some more fiddling.

**Comment by ManoloFLTK** [comment]
No problem here with window resize !
I'm running macOS 13 Ventura, though.

**Comment by ManoloFLTK** [comment]
Unfortunately, I no longer have access to a running macOS 12.

I have built FLTK and gmsh with SDK 11.3 and deployment target=11.0
and did this with git commits abc6709 (still more recent than 4196daa you mentionned) and 59fc60e. I see no problem resizing wndows and no effect of the changes in the FLTK source code.

Is it possible the errors you've seen result from a mix of distinct commits between the libfltk.a used at gmsh link step and FL/*.h include files used at gmsh compile step ?

**Comment by geuzaine** [comment]
I just tested by doing a clean FLTK install but I see the same issue. Here's a short video: can you try the same kind of actions (resize by dragging the middle separator, click on the bottom bar) with your Gmsh build? 

https://user-images.githubusercontent.com/6114922/193462717-93036d75-43fb-4e77-a951-90f58b211219.mov

Basically it seems that the app is missing some redraws, which can be triggered by moving the mouse and waiting for about 1 second.

**Comment by ManoloFLTK** [comment]
No error here (under macOS 13 public beta) when dragging the middle separator or clicking on the bottom bar with the last FLTK commit.

At this point, it would be very helpful if you could identify the commit where your problems appeared.

**Comment by ManoloFLTK** [comment]
I see the problem now here too under macOS 11. I should be able to begin an investigation.

**Comment by ManoloFLTK** [comment]
This issue should be fixed with the last commit (30efed7).
@geuzaine : please close the issue if you confirm.

**Comment by geuzaine** [comment]
@ManoloFLTK: perfect, that fixes it. Thanks!
